include/envoy/stats/scope.h

```diff
@@ -52,6 +53,14 @@ class Scope {
    */
   virtual Counter& counterFromStatName(StatName name) PURE;
 
+  /**
+   * Creates a Counter from the stat name and tags, *without* performing tag extraction on the
+   * stat name.
+   * @param name The name of the stat, obtained from the SymbolTable.
+   * @return a counter within the scope's namespace.
+   */
+  virtual Counter& counterFromStatName(StatName name, const StatNameTagVector& tags) PURE;
```

Comment:
OK I should've though of this before, but WDYT of keeping just one interface per stat-type, and using absl::optional to pass the tag-vector, defaulting to `absl::optiona<StatNameTagVector()`

That would reduce code-duplication in the impls.

Reply:
I like the idea of passing through an optional but I'm a bit iffy on using defaults on virtual functions (I think the google style guide agrees with me here). I'll opt for an explicit overload on the scope virtual class that just invokes it with an empty optional, lmk if that doesnt sound like a good idea

Reply:
works for me if you can get that to compile; I think you might need to use a different method name if signature is implemented in the base class and the other isn't. If that's OK (ie call the new one something different) then I think that's a good solution.

Reply:
Alternatively we could just use the one overload and just require everyone to specify the absl::optional value? It makes the API more explicit but a bit clunkier to use

Reply:
Turns out implementing anything in the interface won't work because StatName is an incomplete class. Giving the explicit signature a shot now.